### ISSUE TITLE
donut-base pretraining learning-rate and batch size

### ISSUE THREAD

**Issue by VictorAtPL** [clovaai/donut#190]
Hi @gwkrsrch,

I need to estimate how long it could take to train donut-base on 16x A100 GPUs.

From _A.5 Implementation and Training Hyperparameters_ of your paper: https://arxiv.org/pdf/2111.15664.pdf

> The initial learning rate of pre-training is set to 1e-4 and that of fine-tuning is selected from 1e-5 to 1e-4. We pre-train the model for 200K steps with 64 NVIDIA A100 GPUs and a mini-batch size of 196, which takes about 2-3 GPU days. We also apply a gradient clipping technique where a maximum gradient norm is selected from 0.05 to 1.0. The input resolution of Donut is set to 2560×1920 at the pre-training phase.

A couple of q's:
1. What do you mean by mini-batch size of 196? 196 / 64 = 3,06. Was 3 or 4 the batch size per GPU?
2. You started per-training with LR set to 1e-4 and what then? Did you use [ReduceLROnPlateau](https://keras.io/api/callbacks/reduce_lr_on_plateau)-like strategy?
3. Your dataset size was 11M of IIT and 2M of SynthDOG, so 13M examples. Each step took 196 examples, so it means one epoch is 66326 steps so you've trained the model for ~3 epochs. Is that correct?
4. Was your A100 GPU a 40GB or 80GB of memory version?

**Comment by gwkrsrch** [comment]
Hi, @VictorAtPL :)

Here are brief answers to your questions. Hope this helps ;) (1) 64*3 is correct, thank you for the question; (2) Yes, we used cosine scheduling and found it to be robust across different scheduling methods; (3) Yes; (4) 80GB.

**Comment by VictorAtPL** [comment]
@gwkrsrch 

Thank you!

1. In pretraining, cosine scheduling like (a), (b) or (c)? I know in fine-tuning in the code there is implemented cosine scheduling with warmup, but did you use the same for pre-training or other method which is not in the code?
a)
![image](https://github.com/clovaai/donut/assets/92857445/db1e69b5-ab6d-4e6d-8e60-dc2825cac5b8)
b) (Tmax = number of all steps)
![image](https://github.com/clovaai/donut/assets/92857445/51ad67d2-07a9-4d9a-a126-45dc37fab167)
c) (Tmax < number of all steps)
![image](https://github.com/clovaai/donut/assets/92857445/7a09196a-2e1b-4bd5-89f8-06862e528324)

2. If the batch size per GPU was 3 and you used 80GB A100 version, no way I can fit 2 examples in 40GB A100 version? I'd have to reduce the resolution of the input image?

**Comment by gwkrsrch** [comment]
1) same code, 2) yes you might need to adjust some hyperparamters, or maybe deepspeed, PyTorch2.0 compile , ... might be helpful to you ;)
